src/hydra/hydraClient.js

```diff
@@ -201,9 +201,19 @@ export default ({entrypoint, resources = []}, httpClient = fetchHydra) => {
         if (page) collectionUrl.searchParams.set('page', page);
         if (perPage) collectionUrl.searchParams.set('perPage', perPage);
         if (params.filter) {
-          Object.keys(params.filter).map(key =>
-            collectionUrl.searchParams.set(key, params.filter[key]),
-          );
+          Object.keys(params.filter).forEach(key => {
+            const filterValue = params.filter[key];
+            if (isPlainObject(filterValue)) {
+              Object.keys(filterValue).forEach(subKey => {
+                collectionUrl.searchParams.set(
+                  `${key}[${subKey}]`,
+                  filterValue[subKey],
+                );
+              });
```

Comment:
You could add a `return` here to avoid the next `if` and reduce a bit the cyclomatic complexity